### ISSUE TITLE
Allow upload of local files.

### ISSUE THREAD

**Issue by WhileLoop** [lambtron/emojipacks#112]
Is this possible? Didn't see anything in the readme about it.

**Comment by zachfeldman** [comment]
Yeah I think you could make a YAML file that just points to files on your local machine, have you tried that?

**Comment by WhileLoop** [comment]
Tried it. Maybe I got the syntax wrong?

Uploading aws with ./aws.png
(node:23728) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): Error: Invalid URI "./aws.png"

Uploading aws with file://./aws.png
(node:23777) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): Error: Invalid protocol: file:

**Comment by zachfeldman** [comment]
Have you tried an absolute path instead of a relative one?

**Comment by WhileLoop** [comment]
Absolute path didn't work either.

**Comment by zachfeldman** [comment]
@WhileLoop Got it. I took a deeper dive into the upload method and realized it loads emojis using the `request()` lib:
https://github.com/lambtron/emojipacks/blob/master/lib/slack.js#L188

In order to load local files, you'd have to integrate a local file reader. Here's a a similar thread:
http://stackoverflow.com/questions/30851520/get-local-json-file-using-request-promise

Alternately, you could upload them somewhere temporary and take them down when you're done - you don't need to commit the .yaml file to the repo to use it!

**Comment by kinduff** [comment]
:+1:

**Comment by EaterOA** [comment]
You can also run a local http server (e.g. `python3 -m http.server`). Then you can use localhost urls.

**Comment by kinduff** [comment]
Nice workaround @EaterOA

**Comment by moustachedelait** [comment]
Any recommendations on how to generate a YAML file of a local directory of images?
